app/controllers/yelp_controller.rb

```diff
@@ -9,11 +9,13 @@ def yelp
   end
 
   def search
-    @address  = params[:address]
-    @term     = params[:term]
-    @response = Yelp.client.search(@address, { term: @term, limit: 20, sort: 1, radius_filter: 8046.72})
-    @response = @response.businesses.to_json
-    render json: @response
+    @address      = params[:address]
+    @vets         = Yelp.client.search(@address, {category_filter: 'vet', limit: 10, sort: 1, radius_filter: 8046.72})
+    @pet_services = Yelp.client.search(@address, {category_filter: 'dogwalkers,pet_sitting,groomer,pet_training', limit: 10, sort: 1, radius_filter: 8046.72})
+    @pet_stores   = Yelp.client.search(@address, {category_filter: 'petstore', limit: 10, sort: 1, radius_filter: 8046.72})
+    @parks        = Yelp.client.search(@address, {category_filter: 'parks,dog_parks,beaches', limit: 10, sort: 1, radius_filter: 8046.72})
+    @response     = @vets.businesses + @pet_services.businesses + @pet_stores.businesses + @parks.businesses
```

Comment:
This is a good opportunity for a `Hash` instead of just lumping them all together in a big array you could create a `Hash` with four keys (`:vets`, `:stores`, etc...) with the list of each as the value. That way you retain the differences in each list and the count of each type.